.github/workflows/publish.yml

```diff
@@ -0,0 +1,65 @@
+name: publish
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+
+  build-test:
+    name: Build-Test
+    strategy:
+      matrix:
+        go-version: [1.15.x]
+        platform: [ubuntu-latest]
+
+    runs-on: ${{ matrix.platform }}
+
+    steps:
+      - name: Set up Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+        id: go
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - name: build
+        run: make build
+      - name: Test
+        run: make test
+
+  publish-derek:
+    runs-on: ubuntu-latest
+    needs: build-test
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Get Repo Owner
+        id: get_repo_owner
+        run: echo ::set-output name=repo_owner::$(echo ${{ github.repository_owner }} | tr '[:upper:]' '[:lower:]')
+      - name: Get TAG
+        id: get_tag
+        run: echo ::set-output name=TAG::${GITHUB_REF#refs/tags/}
+      - name: Login to GHCR Registry
+        uses: docker/login-action@v1
+        with:
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          registry: ghcr.io
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: |
+            ghcr.io/${{ steps.get_repo_owner.outputs.repo_owner }}/derek:${{ steps.get_tag.outputs.TAG }}
```

Comment:
No need to change this now, but if `github.repository_owner` gives the same value as `steps.get_repo_owner.outputs.repo_owner`, then we could probably remove that step in the future. I think `github.repository_owner` must be new, because we didn't find it when looking before? cc @utsavanand2

Reply:
i thought that, but i saw an existing example in another openfaas repo that I followed, but I think we can update all to use the envs the github provide, including the tag

I can open a follow up or if you would like